pkg/controller/controller.go

```diff
@@ -42,10 +42,46 @@ import (
 	"k8s.io/utils/mount"
 )
 
+// RunLocalController facilitates and manages the sync loop.
+// It launches the main sync loop and if there is an updated configuration from the ConfigWatcher,
+// it will inform the main sync loop to terminate and then will launch a new sync loop with the
+// updated configuration.
+func RunLocalController(configUpdate <-chan common.ProvisionerConfiguration, client *kubernetes.Clientset, ptable deleter.ProcTable, discoveryPeriod time.Duration, node *v1.Node, namespace, jobImage string, config common.ProvisionerConfiguration) {
+	stopped := make(chan struct{})
```

Comment:
This pattern of sending a signal and waiting for a callback can be done with a `chan chan err` too as described here: https://go.dev/talks/2013/advconc.slide#25

Reply:
Thanks for the suggestion. Created a new struct called `signal` with `closing chan chan struct{}`

```go
// signal represents an indication to from client to terminate a service and waits for a callback
// indicating that the service has successfully stopped.
type signal struct {
	closing chan chan struct{}
}

func newSignal() *signal {
	return &signal{
		closing: make(chan chan struct{}),
	}
}

func (s *signal) stop() {
	stopped := make(chan struct{})
	s.closing <- stopped
	<-stopped
}

func (s *signal) close() {
	close(s.closing)
}
```

and have `StartLocalController` take in the `*signal` rather than two channels.

This does make the code cleaner, thanks for the suggestion.